### PR TITLE
fix(source-postgres): Support more Postgres and CockroachDB types

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/8922.

Unlike the Postgres destination where we know the column types in advance, in the source plugin we need to account for types that are not part of the SDK schema.

Based on https://www.postgresql.org/docs/current/datatype.html#DATATYPE-TABLE and https://www.cockroachlabs.com/docs/stable/data-types.html#supported-types.

~~I'll update the tests to create a table with each possible type~~ Done

### CODE REVIEW

File: plugins/source/postgresql/resources/plugin/plugin_test.go [hunk 1]
```diff
@@ -52,20 +53,78 @@ func getTestConnectionString() string {
...
 	sb.WriteString("CREATE TABLE ")
-	sb.WriteString(table.Name)
+	sb.WriteString(tableName)
 	sb.WriteString(" (")
-	c := client.Client{}
-	for i, col := range table.Columns {
-		sb.WriteString(col.Name)
+	columns := getPgTypesToData()
+	for i, col := range columns {
+		sb.WriteString(pgx.Identifier{col.typeName + "_type"}.Sanitize())
```
Comment (candiduslynx) [inline plugins/source/postgresql/resources/plugin/plugin_test.go:121]: Why not use column names from DB?

Comment (candiduslynx) [reply plugins/source/postgresql/resources/plugin/plugin_test.go:121]: will create clash on several columns with the same type

File: plugins/source/postgresql/resources/plugin/plugin_test.go [hunk 2]
```diff
@@ -76,33 +135,105 @@ func createTestTable(ctx context.Context, conn *pgxpool.Pool, table *schema.Tabl
...
-		if i < len(table.Columns)-1 {
-			sb.WriteString(", ")
+func insertTestTable(ctx context.Context, conn *pgxpool.Pool, tableName string, columns []pgTypeToValue) error {
+	var query = ""
+	query += "INSERT INTO " + tableName + " ("
+	for _, col := range columns {
+		if col.value == nil {
+			continue
 		}
+		query += pgx.Identifier{col.typeName + "_type"}.Sanitize() + ", "
```
Comment (candiduslynx) [inline plugins/source/postgresql/resources/plugin/plugin_test.go:145]: clash on columns with the same type

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/postgresql/resources/plugin/plugin_test.go (hunk 1 above)
Comment (erezrokah) [reply plugins/source/postgresql/resources/plugin/plugin_test.go:121]: Can you please explain your comment? We don't get the column names from the DB. `columns` have test data generated here https://github.com/cloudquery/cloudquery/pull/8926/files#diff-715416571f0e3a5047731c92c67565cf315764ec01e5b2d59969b00bc99a0b70R61

File: plugins/source/postgresql/resources/plugin/plugin_test.go (hunk 2 above)
Comment (erezrokah) [reply plugins/source/postgresql/resources/plugin/plugin_test.go:145]: See previous column

File: plugins/source/postgresql/resources/plugin/plugin_test.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/postgresql/resources/plugin/plugin_test.go:121]: Ah, I reviewed and didn't notice that this is a test.

File: plugins/source/postgresql/resources/plugin/plugin_test.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/postgresql/resources/plugin/plugin_test.go:145]: Ah, I reviewed and didn't notice that this is a test.

**Review by candiduslynx — APPROVED** [review]
LGTM

**Pull request merged**